source/common/upstream/cluster_manager_impl.cc

```diff
@@ -208,6 +217,15 @@ void ClusterManagerInitHelper::setInitializedCb(std::function<void()> callback)
   }
 }
 
+void ClusterManagerInitHelper::setPrimaryClustersInitializedCb(std::function<void()> callback) {
+  if (state_ == State::WaitingToStartSecondaryInitialization) {
```

Comment:
Should this ASSERT that we're not past the the stage at which this callback would get invoked at?

Reply:
+1 on some stronger ASSERTs around what the valid states here are, as this seems a bit unexpected for a callback setter.

Reply:
Done